buildingsync_asset_extractor/processor.py

```diff
@@ -836,20 +741,14 @@ def format_lighting_results(self, name: str, results: list, units: Optional[str]
             return
 
         # check method 1
-        has_lpd = 1
-        for r in results:
-            try:
-                r['lpd']
-            except Exception:
-                has_lpd = 0
-
+        has_lpd = all([isinstance(r, LightingDataLPD) for r in results])
         # for weighted average, re-find Watts from LPD and LinkedPremises and divide by total sqft
         if has_lpd:
             value = 0.0
             total_sqft = 0
             for r in results:
-                value += r['lpd'] * r['sqft']
-                total_sqft += r['sqft']
+                value += r.lpd * r.sqft
+                total_sqft += r.sqft
             if value > 0:
```

Comment:
catch divide by zero

```suggestion
            if value > 0 and total_sqft != 0:
```